### PR TITLE
Fix wifi candidate selection to include non-interface-bound candidates

### PR DESCRIPTION
**Pull request by Otterverse**
Wifi networks found from scan were associated with the scanning interface (e.g. wlan0) but externally configured wifi settings (e.g. prexisting networks) might not be associated with a specific interface. Thus, duplicate networks (one for "any" and one for "wlan0" were causing issues.

This PR should deduplicate the two maps. If a connection/settings has no interface associated (and it's wifi) it gets handled as though it WAS associated with the current HotspotInterface (can be configured, otherwise first detected wifi adapter.)

### CODE REVIEW

File: subsystems/provisioning/networkmanager.go [hunk 1]
```diff
@@ -607,11 +601,10 @@ func (w *Provisioning) mainLoop(ctx context.Context) {
...
 				cfg := NetworkConfig{
-					Type:      NetworkTypeWifi,
-					SSID:      userInput.SSID,
-					PSK:       userInput.PSK,
-					Priority:  priority,
-					Interface: w.Config().HotspotInterface,
+					Type:     NetworkTypeWifi,
+					SSID:     userInput.SSID,
+					PSK:      userInput.PSK,
+					Priority: priority,
```
Comment (ale7714) [inline subsystems/provisioning/networkmanager.go:607]: why is interface not needed?

Comment (ale7714) [reply subsystems/provisioning/networkmanager.go:607]: nvm i see bellow

**Review by ale7714 — APPROVED** [review]

**Review by ale7714 — APPROVED** [review]

File: cmd/viam-agent/main.go [hunk 2]
```diff
@@ -183,7 +183,7 @@ func main() {
 	}
 
 	// wait until now when we (potentially) have a network logger to record this
-	globalLogger.Infof("Viam Agent Version: %s\nGit Revision: %s\n", agent.GetVersion(), agent.GetRevision())
+	globalLogger.Infof("Viam Agent Version: %s Git Revision: %s", agent.GetVersion(), agent.GetRevision())
```
Comment (Otterverse) [inline cmd/viam-agent/main.go:186]: drive-by... didn't realize I'd copy/pasted the newlines from the CLI "version" output here (where it makes no sense.)
